libs/components/popovers/testing/src/dropdown/harness/dropdown-menu-harness.ts

```diff
@@ -0,0 +1,116 @@
+import { HarnessPredicate } from '@angular/cdk/testing';
+import { SkyComponentHarness } from '@skyux/core/testing';
+
+import { SkyDropdownItemHarness } from './dropdown-item-harness';
+import { SkyDropdownMenuHarnessFilters } from './dropdown-menu-harness.filters';
+
+/**
+ * Harness for interacting with dropdown menu in tests
+ */
+export class SkyDropdownMenuHarness extends SkyComponentHarness {
+  /**
+   * @internal
+   */
+  public static hostSelector = '.sky-dropdown-menu';
+
+  #getDropdownItems = this.locatorForAll(SkyDropdownItemHarness);
+
+  /**
+   * Gets a `HarnessPredicate` that can be used to search for a
+   * `SkyDropdownMenuHarness` that meets certain criteria
+   */
+  public static with(
+    filters: SkyDropdownMenuHarnessFilters
+  ): HarnessPredicate<SkyDropdownMenuHarness> {
+    return SkyDropdownMenuHarness.getDataSkyIdPredicate(filters);
+  }
+
+  /**
+   * Gets the aria-labelledby value
+   */
+  public async getAriaLabelledBy(): Promise<string | null> {
+    return (await this.host()).getAttribute('aria-labelledby');
+  }
+
+  /**
+   * Clicks the dropdown menu item at index
+   * @param index Index of dropdown menu item
+   */
+  public async clickItemAtIndex(index: number): Promise<void> {
```

Comment:
For all of these methods interacting with dropdown items, I would take a look at what we did for autocomplete. Essentially, you would provide the consumer with a `getMenuItems` method with a filtering mechanism, and a `clickMenuItem` method (also with a filtering mechanism). The consumer can then call the item harness's methods directly in their tests (such as clicking an item, or getting its `ariaRole`), removing the need to have several of the item-specific methods below.

Autocomplete's `getSearchResults` could be tailored to get menu items: https://github.com/blackbaud/skyux/blob/main/libs/components/lookup/testing/src/autocomplete/autocomplete-harness.ts#L143C5-L150

Autocomplete getting search results using a filter: https://github.com/blackbaud/skyux/blob/main/libs/components/lookup/testing/src/autocomplete/autocomplete-harness.spec.ts#LL158C5-L164C7

Autocomplete's `selectSearchResults` could be reused to click a specific menu item: https://github.com/blackbaud/skyux/blob/main/libs/components/lookup/testing/src/autocomplete/autocomplete-harness.ts#L143-L150